src/main/java/net/minecraftforge/fml/common/IEntitySelectorFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fml.common;
+
+import com.google.common.base.Predicate;
+import net.minecraft.command.EntitySelector;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.Vec3d;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Can be registered in {@link net.minecraftforge.fml.common.registry.GameRegistry} to generate entity predicates for custom entity selectors in commands.
```

Comment:
This should contain the most important information first. It should also point directly to the registration method, GameRegistry is a large file to look through. Something like this:

```java
/**
 * Allows mods to create custom entity selectors in commands.
 * Registered in {@link net.minecraftforge.fml.common.registry.GameRegistry#registerEntitySelector}.
 * For an example implementation, see CustomEntitySelectorTest.
```